server/src/main/java/org/elasticsearch/cluster/metadata/SingleNodeShutdownMetadata.java

```diff
@@ -0,0 +1,268 @@
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.cluster.AbstractDiffable;
+import org.elasticsearch.cluster.Diffable;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentFragment;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Contains data about a single node's shutdown readiness.
+ */
+public class SingleNodeShutdownMetadata extends AbstractDiffable<SingleNodeShutdownMetadata>
+    implements
+        ToXContentObject,
+        Diffable<SingleNodeShutdownMetadata> {
+
+    private static final ParseField NODE_ID_FIELD = new ParseField("node_id");
+    private static final ParseField TYPE_FIELD = new ParseField("type");
+    private static final ParseField REASON_FIELD = new ParseField("reason");
+    private static final ParseField STATUS_FIELD = new ParseField("status");
+    private static final String STARTED_AT_READABLE_FIELD = "shutdown_started";
+    private static final ParseField STARTED_AT_MILLIS_FIELD = new ParseField(STARTED_AT_READABLE_FIELD + "millis");
+    private static final ParseField SHARD_MIGRATION_FIELD = new ParseField("shard_migration");
+
+    public static final ConstructingObjectParser<SingleNodeShutdownMetadata, Void> PARSER = new ConstructingObjectParser<>(
+        "node_shutdown_info",
+        a -> new SingleNodeShutdownMetadata((String) a[0], (String) a[1], (String) a[2], Status.valueOf((String) a[3]), (long) a[4],
+            (ComponentShutdownStatus) a[5])
+    );
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), NODE_ID_FIELD);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), TYPE_FIELD);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), REASON_FIELD);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), STATUS_FIELD);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), STARTED_AT_MILLIS_FIELD);
+        PARSER.declareObject(
+            ConstructingObjectParser.constructorArg(),
+            (parser, context) -> ComponentShutdownStatus.parse(parser),
+            SHARD_MIGRATION_FIELD
+        );
+    }
+
+    public static SingleNodeShutdownMetadata parse(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    private final String nodeId;
+    private final String type;
+    private final String reason;
+    private final Status status;
+    private final long startedAtDate;
```

Comment:
Can you change this to include the unit type, since we can't tell its resolution from the name (something like `startedMillis` or `startedNanos` depending on the resolution)